cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/GoalOptimizer.java

```diff
@@ -117,6 +117,52 @@ public GoalOptimizer(KafkaCruiseControlConfig config,
     _proposalComputationTimer = dropwizardMetricRegistry.timer(MetricRegistry.name("GoalOptimizer", "proposal-computation-timer"));
   }
 
+  /**
+   * Generate unique goal priority for each of proposal precomputing thread.
+   */
+  private void populateGoalByPriorityForPrecomputing() {
+    List<List<Goal>> shuffledGoalList = new ArrayList<>();
```

Comment:
Do we need a list of list of goals or would it suffice to have a set of list of goals?